### PR TITLE
feat: toggle button group to select severity results to display

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

On the Image Checker screen, adds toggle buttons to filter results by severity.

TODO
- [x] add unit tests

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/9973512/33c17817-11cb-4816-bb56-df41535a5688

### What issues does this PR fix or reference?

#4985

### How to test this PR?

You can install a fake image checker from the image at quay.io/phmartin/fake-image-checker:v0.0.3 (sources: https://github.com/feloy/podman-desktop-extension-fake-image-checker)

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
hi, my only comment would be about the appearance of the toggles

we see toggle even when we're still processing checkers and we don't have any result.

so I don't know if we should hide toggles until we have checkers/results or disable toggles (like greyed colors) so we see them but it's not really enabled or if there is a missing loading/spinner/horizontal bar in the right part of the screen let me think that there is something that might appear.

As a side note, if all checkers run successfully but do not report anything (no warning, no success), do we see something on the right or it's blank ?

**Comment by feloy** [comment]
> hi, my only comment would be about the appearance of the toggles
> 
> we see toggle even when we're still processing checkers and we don't have any result.
> 
> so I don't know if we should hide toggles until we have checkers/results or disable toggles (like greyed colors) so we see them but it's not really enabled or if there is a missing loading/spinner/horizontal bar in the right part of the screen let me think that there is something that might appear.
> 
> As a side note, if all checkers run successfully but do not report anything (no warning, no success), do we see something on the right or it's blank ?

When the processing is still running, the toggles are actually disabled (as a side effect that the count is zero for the different severities). 

When a first provider has finished running, the buttons start to be enabled, depending on the severities available in the result.

**Comment by benoitf** [comment]
in the recorded video we still see
image

and toolbar does not render like 'disabled' to me

**Comment by feloy** [comment]
I accentuated the disabled state, and added the colors on icons I has forgotten (only when not disabled)

**Comment by benoitf** [comment]
thanks @feloy looks like much better

**Review by benoitf — APPROVED** [review]
I thought I approved it this morning but it seems no...

**Review by cdrage — APPROVED** [review]
Screenshot 2023-11-27 at 9 15 38 PM

Sorry was late with the review. But the code LGTM and I was able to test it. The toggling looks good.

One small comment, but I honestly don't think it matters.

File: packages/renderer/src/lib/ui/ProviderResultPage.svelte [hunk 1]
```diff
@@ -53,20 +88,75 @@ function onProviderChecked(id: string, checked: boolean) {
...
+      return selectedSeverities['success'];
+    }
+    if (r.check.severity) {
+      return selectedSeverities[r.check.severity];
+    }
+    return true;
+  });
+}
+
+function onSeverityClicked(severity: 'critical' | 'high' | 'medium' | 'low' | 'success', clicked: boolean) {
```
Comment (cdrage) [inline packages/renderer/src/lib/ui/ProviderResultPage.svelte:107]: Should `'critical' | 'high' | 'medium' | 'low' | 'success'` be a const / type instead?

But honestly not a big deal if we change it. Everything LGTM code-wise!

**Pull request merged**